### PR TITLE
Relative paths in noparse option

### PR DESCRIPTION
**Pull request by goldim**
Browserify command with relative paths via `--noparse` option doesn't work. If I pass full paths in the option everything works as expected. Digging the problem I noticed that `noparse` array are passed into `module-deps` class via constructor. And in this package there is no processing the array like in browserify package (basedir+cwd+file name in noparse array).

### CODE REVIEW

**Comment by mikixing** [comment]
您的邮件我已收到，会尽快查阅~

**Comment by ljharb** [comment]
This will need a regression test.

**Comment by goldim** [comment]
@ljharb Hello, you mean new tests/test in which I have to check passing relative paths for `noParse` argument, don't you? Maybe I've hurried creating this PR. I think an issue first would be good idea. What do you think?
Also I've found this (issue)[https://github.com/browserify/browserify/issues/1350] in which the author complained about not working command browserify and he passed relative paths. In comments someone told him that paths have to be absolute. 
But nowhere is said that paths have to be absolute.

**Comment by ljharb** [comment]
I mean a test that would fail without the changes in this PR.

**Comment by goldim** [comment]
@ljharb Done.

File: test/noparse.js
Comment (ljharb) [inline test/noparse.js:44]:
```suggestion
    ].map(function (x) { return path.resolve(x); }).sort();
```

File: test/noparse.js
Comment (ljharb) [inline test/noparse.js:53]:
```suggestion
    b.on('dep', function (dep) { actual.push(dep.file); });
```

File: test/noparse.js
Comment (ljharb) [inline test/noparse.js:47]:
```suggestion
        entries: [__dirname + '/noparse/a.js'],
```

**Review by ljharb — APPROVED** [review]
Assuming that the test fails without your fix, and passes with it, this LGTM modulo nits

**Comment by mikixing** [comment]
您的邮件我已收到，会尽快查阅~

**Comment by goto-bus-stop** [comment]
Released in 📦 17.0.1. Thank you!

**Pull request merged**
